### ISSUE TITLE
Failed to allocate device vector A (error code unknown error)!

### ISSUE THREAD

**Issue by FlorinAndrei** [NVIDIA/gpu-operator#854]
### 1. Quick Debug Information
* OS/Version(e.g. RHEL8.6, Ubuntu22.04): Ubuntu 22.04.2
* Kernel Version: 5.15.0-101-generic
* Container Runtime Type/Version(e.g. Containerd, CRI-O, Docker): Containerd
* K8s Flavor/Version(e.g. K8s, OCP, Rancher, GKE, EKS): MicroK8s v1.27.13 revision 6743
* GPU Operator Version: 22.9.1
* NVIDIA driver version installed at the host level: 535.104.12 (not using the operator's own drivers)

### 2. Issue or feature description
4 node cluster, bare metal. One worker node was rebooted for maintenance. When it came back up, the operator failed to run properly on it. No GPU workloads could be scheduled on this node.

### 3. Steps to reproduce the issue
See above.

### 4. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [x] kubernetes pods status: `kubectl get pods -n OPERATOR_NAMESPACE`
 - [x] kubernetes daemonset status: `kubectl get ds -n OPERATOR_NAMESPACE`
 - [x] If a pod/ds is in an error state or pending state `kubectl describe pod -n OPERATOR_NAMESPACE POD_NAME`
 - [x] If a pod/ds is in an error state or pending state `kubectl logs -n OPERATOR_NAMESPACE POD_NAME --all-containers`
 - [x] Output from running `nvidia-smi` from the driver container: `kubectl exec DRIVER_POD_NAME -n OPERATOR_NAMESPACE -c nvidia-driver-ctr -- nvidia-smi`
 - [x] containerd logs `journalctl -u containerd > containerd.log`

Collecting full debug bundle (optional):

```
curl -o must-gather.sh -L https://raw.githubusercontent.com/NVIDIA/gpu-operator/master/hack/must-gather.sh 
chmod +x must-gather.sh
./must-gather.sh
```
**NOTE**: please refer to the [must-gather](https://raw.githubusercontent.com/NVIDIA/gpu-operator/master/hack/must-gather.sh) script for debug data collected.

This bundle can be submitted to us via email: **operator_feedback@nvidia.com**

**Comment by FlorinAndrei** [comment]
NVIDIA drivers are installed at the host level. No drivers are installed by the operator.

Here's the information to attach:

[kubectl_output.txt](https://github.com/user-attachments/files/16316261/kubectl_output.txt)

I will send the full debug bundle to the email address.

**Comment by cdesiniotis** [comment]
Hi @FlorinAndrei would you be able to run `nvidia-bug-report.sh` on the problematic node and share that with us as well? I noticed the must-gather script did not collect the `nvidia-bug-report` since your drivers are pre-installed.

**Comment by FlorinAndrei** [comment]
nvidia bug report:

[nvidia-bug-report.log.gz](https://github.com/user-attachments/files/16317265/nvidia-bug-report.log.gz)

@cdesiniotis To add: this node was working fine before I rebooted it. It has rebooted into the same kernel and Nvidia driver versions as before. The other 3 nodes in the same k8s cluster are still working just fine right now (but no reboots there).

**Comment by FlorinAndrei** [comment]
@cdesiniotis Please ignore the compile error from nvidia-bug-report. That seems to be for another kernel installed here. We're not running that kernel.

**Comment by cdesiniotis** [comment]
I see the following messages in the system log:
```
Jul 19 00:58:52 GPUF061 systemd-udevd[1181]: nvidia: Process '/sbin/modprobe nvidia-modeset' failed with exit code 1.
Jul 19 00:58:52 GPUF061 systemd-udevd[1181]: nvidia: Process '/sbin/modprobe nvidia-drm' failed with exit code 1.
Jul 19 00:58:52 GPUF061 systemd-udevd[1181]: nvidia: Process '/sbin/modprobe nvidia-uvm' failed with exit code 1.
```
Can you verify whether these modules are loaded?

**Comment by FlorinAndrei** [comment]
Current status:

```
root@GPUF061:~# lsmod | grep ^nvidia
nvidia_drm             77824  0
nvidia_modeset       1302528  1 nvidia_drm
nvidia              56619008  496 nvidia_modeset
```

However:

```
root@GPUF061:~# modprobe nvidia_drm
modprobe: FATAL: Module nvidia_drm not found in directory /lib/modules/5.15.0-101-generic
root@GPUF061:~# modprobe nvidia_modeset
modprobe: FATAL: Module nvidia_modeset not found in directory /lib/modules/5.15.0-101-generic
root@GPUF061:~# modprobe nvidia
modprobe: FATAL: Module nvidia not found in directory /lib/modules/5.15.0-101-generic
```

Not sure how to reconcile those two.

**Comment by FlorinAndrei** [comment]
@cdesiniotis  On the broken system:

```
root@GPUF061:~# lsmod | grep nvidia
nvidia_drm             77824  0
nvidia_modeset       1302528  1 nvidia_drm
nvidia              56619008  496 nvidia_modeset
drm_kms_helper        311296  4 mgag200,nvidia_drm
drm                   622592  5 drm_kms_helper,nvidia,mgag200,nvidia_drm
root@GPUF061:~# modprobe nvidia_drm
modprobe: FATAL: Module nvidia_drm not found in directory /lib/modules/5.15.0-101-generic
root@GPUEFB1:~# 
```

On a good system:

```
root@GPUEFB1:~# lsmod | grep nvidia
nvidia_uvm           1511424  148
nvidia_drm             77824  0
nvidia_modeset       1302528  1 nvidia_drm
nvidia              56659968  5462 nvidia_uvm,nvidia_modeset
drm_kms_helper        311296  4 mgag200,nvidia_drm
drm                   622592  5 drm_kms_helper,nvidia,mgag200,nvidia_drm
root@GPUEFB1:~# modprobe nvidia_drm
root@GPUEFB1:~# 
```

**Comment by FlorinAndrei** [comment]
I've figured it out. The Nvidia driver was in this weird, partially loaded state on the broken system. Only some modules were loaded, but not all. No idea why. It looked good in the system-level checks I did before, but obviously it was not in good health.

I've re-run `dpkg-reconfigure nvidia-dkms-535` on the broken system, and it has re-compiled the Nvidia modules for the running kernel, and installed them. Now I see them all.

After that, the GPU operator has "magically" recovered. Now the cluster is scheduling GPU workloads on this node. Everything looks good.

Thank you for your help, and sorry for the false alarm.
